src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs

```diff
@@ -112,12 +113,21 @@ public void Abort()
         {
             if (_disposed)
             {
-                throw new ObjectDisposedException("DiscoveryRequest");
+                throw new ObjectDisposedException(nameof(DiscoveryRequest));
             }
 
             if (DiscoveryInProgress)
             {
-                DiscoveryManager.Abort();
+                // If testhost has old version, we should use old cancel logic
+                // to be consistent and not create regression issues
+                if (_protocolConfig.Version < Constants.MinimumProtocolVersionWithCancelDiscoveryEventHandlerSupport)
```

Comment:
I have moved the logic of comparing the version and deciding which abort to call here, I hope that's ok? This way I can get rid of a bunch of intermediate public apis.

Reply:
Ok for me.